### DISCUSSION TITLE
MSO24: Slow curve_query() results

### DISCUSSION THREAD

**Discussion by phxucla** [tektronix/tm_devices#356] [Q&A] [answered]
Reading a waveform from an MSO24 scope using scope.curve_query() is extremely slow. It takes 1 second to read 4 channels of 200 samples each. How can I speed it up?

**Comment by amccann-Tek** [comment, marked as answer]
Hi @phxucla, Sorry for the late reply we are catching up from the holidays.

In these discussion boards we often don't have the deep technical instrument knowledge to assist with topics outside of the python package mechanics and code, but I did find one of our MSO2 folks to chat with on this. It sounds like your experience is roughly within the average transfer times.

A couple things you could try:
- Use only the commands necessary (ex, configure then call scope.commands.curve.query()). the scope.curve_query() method does some extra actions that could be slowing the scope.
- For single shot transfers CURVE? is going to the way to go, but for other types you might look into CURVESTREAM? (scope.commands.curvestream.query())

We are investigating if we can bring our HSI (and TekHSI) technology onto the 2-Series for faster transfer rates but that work is ongoing.

If you are willing to provide some more details around your work/application I think that would help us improve in future updates.

If you have other questions about the scope itself I would reach out to our Technical Support Center - https://www.tek.com/en/support
 

Regards,
-Andrew
